### PR TITLE
2 Bug Fixes/Feature Addition

### PR DESCRIPTION
**Pull request by gokturkDev**
Bug #1 state needs to be passed to event handlers. Otherwise they cannot manipulate the state. And since grand magnificent js does monkey business with objects, trying to acces the bloc state from funcs is hard, error prone and hurts readability. 

Bug/Feature #2 We need to be able to pass in instantiated events. Earlier this was not possible due to our lord and savior js and its objects, now still its a hacky solution but atleast we can pass payloads with events

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -25,6 +25,9 @@
   "include": [
     "./src",
     "node_modules/@types/jest/index.d.ts",
+    "node_modules/@types/jest/index.d.ts",
+    "node_modules/@types/jest/index.d.ts",
+    "node_modules/@types/jest/index.d.ts"
```
Comment (jsnanigans) [inline tsconfig.json:30]: this is actually already included above, did adding it three more times fix something for you?
I would not be very surprised if it did 🙃

**Review by jsnanigans — CHANGES_REQUESTED** [review]
This is awesome, thank you for taking the time to find and fix the bugs!
I really appreciate help on the "Bloc" since I mostly only use the "Cubit"

I noticed you like to use `var` or `let` but I prefer to use `const` if the variable does not change. But for the tests it's no big deal

**Comment by gokturkDev** [comment]
My pleasure!, BLoc and BLac to the moon 🚀. Yeah, i use statiscally typed languages like java and dart more, const var let stuff is kind of unorthodox the least for me :d.

File: src/lib/usecase-tests/login/test/login_state.test.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  })
+
+
+  function createLoginState() : LoginState{
+    return new LoginState( new User(
+        "and btw",
+        "blac is awesome"
+      ),
+      new Password(
+        "you did a great job ;)",
```
[the remark this answers is not in the document]
Comment (gokturkDev) [reply src/lib/usecase-tests/login/test/login_state.test.ts:28]: Blac forever 🥇

**Review by gokturkDev — COMMENTED** [review]
I guess this looks fine!

**Review by jsnanigans — APPROVED** [review]
Its very interesting to see how you do everything with classes, I guess thats its normally done in Java and Dart. Thanks again for the help!

**Comment by jsnanigans** [comment]
@gokturkDev your contribution was released in 0.4.1 🎉

**Comment by gokturkDev** [comment]
🙌

**Pull request merged**